src/test/java/de/tum/cit/aet/artemis/iris/PyrisConcistencyCheckIntegrationTest.java

```diff
@@ -0,0 +1,134 @@
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Profile;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.test.context.support.WithMockUser;
+
+import de.tum.cit.aet.artemis.core.domain.Course;
+import de.tum.cit.aet.artemis.core.domain.LLMRequest;
+import de.tum.cit.aet.artemis.core.user.util.UserUtilService;
+import de.tum.cit.aet.artemis.iris.service.IrisConsistencyCheckService;
+import de.tum.cit.aet.artemis.iris.service.pyris.PyrisDTOService;
+import de.tum.cit.aet.artemis.iris.service.pyris.dto.consistencyCheck.PyrisConsistencyCheckStatusUpdateDTO;
+import de.tum.cit.aet.artemis.iris.service.pyris.dto.data.PyrisProgrammingExerciseDTO;
+import de.tum.cit.aet.artemis.iris.service.pyris.dto.status.PyrisStageDTO;
+import de.tum.cit.aet.artemis.iris.service.pyris.dto.status.PyrisStageState;
+import de.tum.cit.aet.artemis.iris.service.pyris.job.ConsistencyCheckJob;
+import de.tum.cit.aet.artemis.programming.domain.ProgrammingExercise;
+import de.tum.cit.aet.artemis.programming.domain.ProgrammingLanguage;
+
+@Profile(PROFILE_IRIS)
+class PyrisConsistencyCheckIntegrationTest extends AbstractIrisIntegrationTest {
+
+    private static final String TEST_PREFIX = "pyrisconsistencytest";
+
+    @Autowired
+    private UserUtilService userUtilService;
+
+    @Mock
+    private PyrisDTOService pyrisDTOService;
+
+    @Autowired
+    private IrisConsistencyCheckService irisConsistencyCheckService;
+
+    private Course course;
+
+    private ProgrammingExercise programmingExercise;
+
+    @BeforeEach
+    void initTestCase() {
+        userUtilService.addUsers(TEST_PREFIX, 2, 1, 0, 2);
+        course = programmingExerciseUtilService.addCourseWithOneProgrammingExercise();
+        this.programmingExercise = course.getExercises().stream().filter(ProgrammingExercise.class::isInstance).map(ProgrammingExercise.class::cast).findFirst().orElse(null);
+
+        userUtilService.createAndSaveUser(TEST_PREFIX + "student42");
+        userUtilService.createAndSaveUser(TEST_PREFIX + "tutor42");
+        userUtilService.createAndSaveUser(TEST_PREFIX + "instructor42");
+        activateIrisFor(course);
+    }
+
+    @Test
+    @WithMockUser(username = TEST_PREFIX + "instructor1", roles = "INSTRUCTOR")
+    void callConsistencyCheckAsInstructor_shouldSucceed() throws Exception {
+        PyrisProgrammingExerciseDTO exerciseDto = createPyrisProgrammingExerciseDTO(programmingExercise);
+        irisRequestMockProvider.mockProgrammingConsistencyCheckResponse(dto -> {
+            assertThat(dto.exercise().id()).isEqualTo(exerciseDto.id());
+            assertThat(dto.exercise().name()).isEqualTo("Programming");
+            assertThat(dto.exercise().programmingLanguage()).isEqualTo(ProgrammingLanguage.JAVA);
+        }, exerciseDto.id());
```

Comment:
I do not understand this, we are testing our mock here - this was quite confusing to me and took me quite some time to understand what is happening here.

If that is really relevant, we should write a test for the shared method with some cases, but I do not see a point of making the assertions in each test.

Reply:
i can reomve it. I just thought, since i have this consumer anyways, it might be usefull to check wheter our mock works as expected